### PR TITLE
Guidelines for range related properties

### PR DESCRIPTION
**Pull request by spectranaut**
This PR is to get feedback on range related property section to add to the APG.

***
<a href="https://pr-preview.s3.amazonaws.com/bocoup/aria-practices/pull/1000.html" title="Last updated on Aug 30, 2019, 1:54 PM UTC (d4a09ee)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/aria-practices/1000/e653a7a...bocoup:d4a09ee.html" title="Last updated on Aug 30, 2019, 1:54 PM UTC (d4a09ee)">Diff</a>

### CODE REVIEW

File: range_related_properties.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
+# Range Related Properties
+
+The range related properties are used to fully present and describe an element whose value can be one of a range of values to assistive technologies. These properties communicate the maximum and minimum values for the element and the element's current value.
+
+The range properties can also be used to fully present and describe an element whose value can be one of a list of non-numeric values. In this scenario, all possible text values of the element will be programmatically mapped to numbers within the numeric range.
```
Comment (spectranaut) [inline range_related_properties.md]: "In this scenario, all possible text values of the element will be programmatically mapped to numbers within the numeric range."

I feel like for this sentence it is not clear who is programmatically doing the mapping -- the answer is that the user who is adding the role should. I'm not sure how to say this -- how do I refer to this user?

File: range_related_properties.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+* `spinbutton`
+* `progressbar`
+* `meter`
+* `scrollbar`
+
+These proprieties are also used for the `separator` role, but only if the element is focusable.
+
+Range Related Property Table
+
+| Property | Definition |
```
Comment (zcorpan) [inline range_related_properties.md]: Move this table up (comment from @mcking65)

File: range_related_properties.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+| --- | --- |
+| `aria-valuemin` | Defines the minimum value in the range. |
+| `aria-valuemax` | Defines the maximum value in the range. |
+| `aria-valuenow` | Defines the value of the element. This value is always a number between `aria-valuemin` and `aria-valuemax`. |
+| `aria-valuetext` | Defines a description of the value of the element. |
+
+## Using `aria-valuemin` and `aria-valuemax`
+
+When an element's possible values are contained within a known range, the attributes `aria-valuemin` and `aria-valuemax` are used to inform assistive technologies of the minimum and maximum values of the range. When using these properties, always set `aria-valuemin` to the lowest value of the range.
```
Comment (zcorpan) [inline range_related_properties.md]: should say what to set `aria-valuemax` to also

File: range_related_properties.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+These proprieties are also used for the `separator` role, but only if the element is focusable.
+
+Range Related Property Table
+
+| Property | Definition |
+| --- | --- |
+| `aria-valuemin` | Defines the minimum value in the range. |
+| `aria-valuemax` | Defines the maximum value in the range. |
+| `aria-valuenow` | Defines the value of the element. This value is always a number between `aria-valuemin` and `aria-valuemax`. |
```
Comment (zcorpan) [inline range_related_properties.md]: Avoid using "always" and "never"

File: range_related_properties.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+| `aria-valuetext` | Defines a description of the value of the element. |
+
+## Using `aria-valuemin` and `aria-valuemax`
+
+When an element's possible values are contained within a known range, the attributes `aria-valuemin` and `aria-valuemax` are used to inform assistive technologies of the minimum and maximum values of the range. When using these properties, always set `aria-valuemin` to the lowest value of the range.
+
+## Using `aria-valuenow`
+
+The attribute `aria-valuenow` is used to inform assistive technologies of the current value of an element. When `aria-valuemin` and `aria-valuemax` are specified, set `aria-valuenow` to a numeric value that is within the range define by `aria-valuemin` and `aria-valuemax`.
```
Comment (zcorpan) [inline range_related_properties.md]: Say that valuenow is allowed also when the range is not known

File: range_related_properties.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+## Using `aria-valuetext`
+
+When the element's values are contained within a range but those values are not numeric (such as "small", "medium" and "large"), `aria-valuetext` is used to surface the text value to assistive technologies. Only use `aria-valuetext` when `aria-valuenow` does not have meaning for the user because using `aria-valuetext` will prevent assistive technologies from communicating `aria-valuenow`.
+
+## `slider` Role
+
+`aria-valuenow`, `aria-valuemin` and `aria-valuemax` are all required attributes for the `slider` role. `aria-valuetext` can be used when appropriate. Detailed description of the `slider` role can be found in the [slider design pattern](https://w3c.github.io/aria-practices/#slider) and [slider (multi-thumb) design pattern](https://w3c.github.io/aria-practices/#slidertwothumb).
+
+### Example with numeric range
```
Comment (zcorpan) [inline range_related_properties.md]: Have examples inline without explicit subsection

File: range_related_properties.md (hunk 1 above)
Comment (zcorpan) [reply range_related_properties.md]: author

File: range_related_properties.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+## `progressbar` Role
+
+`aria-valuenow`, `aria-valuemin` and `aria-valuemax` are not required attributes for the `progressbar` role, however, but the attributes might be necessary for communicating the state of a progress bar to assistive technologies.
+
+This is an example of a progress bar represent by an SVG. The range properties are necessary to full explain the widget to assistive technologies.
+
+```
+<div for="loadstatus">Loading:
+  <span role="progressbar" id="loadstatus" aria-valuemin="0" aria-valuenow="33" aria-valuemax="100" >
```
Comment (zcorpan) [inline range_related_properties.md]: Maybe show how `aria-valuemin` and `aria-valuemax` can be omitted in this example, and that they have default values.

**Comment by zcorpan** [comment]
I have converted this to HTML and rebased on master, but no new changes since @spectranaut had already addressed review feedback.

If this now looks OK, I think it should be ready for merge.

Here's a link to the [preview of the new section](https://pr-preview.s3.amazonaws.com/bocoup/aria-practices/pull/1000.html#range_related_properties).

**Comment by zcorpan** [comment]
@jongund can you review this? Thanks!

File: aria-practices.html [hunk 8]
```diff
@@ -6279,6 +6279,137 @@ <h3>Indicating sort order with <code>aria-sort</code></h3>
 
   </section>
 
+  <section id="range_related_properties">
+    <h2>Range Related Properties</h2>
```
Comment (zcorpan) [inline aria-practices.html]: Change the title to better describe what the range concept is about.

File: aria-practices.html [hunk 9]
```diff
@@ -6279,6 +6279,137 @@ <h3>Indicating sort order with <code>aria-sort</code></h3>
...
+      <li><code>progressbar</code></li>
+      <li><code>meter</code></li>
+      <li><code>scrollbar</code></li>
+      <li><code>separator</code> (if the element is focusable)</li>
+    </ul>
+
+    <section id="range_related_properties_using_aria-valuemin_and_aria-valuemax">
+      <h3>Using <code>aria-valuemin</code> and <code>aria-valuemax</code></h3>
+      <p>When an element's possible values are contained within a known range, the attributes <code>aria-valuemin</code> and <code>aria-valuemax</code> are used to inform assistive technologies of the minimum and maximum values of the range. When using these properties, set <code>aria-valuemin</code> to the lowest value of the range and <code>aria-valuemax</code> to the greatest value.</p>
+      <p>When the range is unknown, omit both <code>aria-valuemin</code> and <code>aria-valuemax</code>. See an example of a indeterminate range element <a href="">indeterminate progress bar</a>.</p>
```
Comment (zcorpan) [inline aria-practices.html]: Fill in `a href` value or remove

**Comment by zcorpan** [comment]
Replacing this PR with #1279

**Pull request closed without merging**